### ISSUE TITLE
FlyingSaucer can cleanup invalid html

### ISSUE THREAD

**Issue by asolntsev** [flyingsaucerproject/flyingsaucer#279]
FlyingSaucer could be more tolerant to minor errors in html source. 
Instead of throwing an exception, say, for non-closed html5 tags, it could close them automatically.

For example, we could use JSoup for job. 

Triggered in discussion https://github.com/flyingsaucerproject/flyingsaucer/pull/277

**Comment by DarkTyger** [comment]
I agree with @pbrant : If an application wants to clean up HTML before passing it to FlyingSaucer, that's the application's job. A module would be good, but I already have a specific version of JSoup included with my application for various reasons (including, but not limited to, clean up before passing to Flying Saucer). Having potentially two versions of JSoup would unnecessarily bloat my application.

Can we close this issue?

**Comment by asolntsev** [comment]
@DarkTyger Yes, we could close the issue and leave this problem for FS users to solve by themselves.
But it's a common request, and many users want to solve this problem.
So it's better to solve it once in FS than make people re-invent the wheel every time.

> An optional plugin (or module) would be good.

This is exactly what I was think about: a separate plugin / module / something like that. 

> Having potentially two versions of JSoup would unnecessarily bloat my application.
You will never have two versions of JSoup. Both Maven and Gradle always leave only one version in classpath.
